src/mmw/js/src/data_catalog/views.js

```diff
@@ -126,20 +131,84 @@ var FormView = Marionette.ItemView.extend({
     className: 'data-catalog-form',
 
     ui: {
+        dateInput: '.data-catalog-date-input',
+        filterToggle: '.date-filter-toggle',
         searchInput: '.data-catalog-search-input'
     },
 
+    modelEvents: {
+        'change:showingFilters change:isValid': 'render'
+    },
+
     events: {
-        'keyup @ui.searchInput': 'onSearchInputChanged'
+        'keyup @ui.searchInput': 'onSearchInputChanged',
+        'click @ui.filterToggle': 'onFilterToggle',
+        'change @ui.dateInput': 'onDateInputChanged',
+        'keyup @ui.dateInput': 'onDateInputKeyup'
+    },
+
+    onRender: function() {
+        $('.data-catalog-date-input').datepicker();
     },
 
     onSearchInputChanged: function(e) {
         var query = this.ui.searchInput.val().trim();
         if (e.keyCode === ENTER_KEYCODE) {
-            this.triggerMethod('search');
+            this.triggerSearch();
         } else {
             this.model.set('query', query);
         }
+    },
+
+    onDateInputKeyup: function(e) {
+        if (e.keyCode === ENTER_KEYCODE) {
+            this.triggerSearch();
+        }
+    },
+
+    onDateInputChanged: function(e) {
+        var dateControl = $(e.currentTarget),
+            isStartDate = dateControl.data('dateRange') === 'start',
+            attr = isStartDate ? 'dateStart' : 'dateEnd';
+
+        this.model.set(attr, dateControl.val());
+    },
+
+    onFilterToggle: function() {
+        var newVal = !this.model.get('showingFilters');
+        this.model.set('showingFilters', newVal);
+    },
+
+    triggerSearch: function() {
+        if (this.validate()) {
+            this.triggerMethod('search');
+        }
+    },
+
+    validate: function() {
+        // Only need to validate if there are two dates.  Ensure that
+        // before is earlier than after
+        var dateFormat = "MM/DD/YYYY",
+            before = moment(this.model.get('dateStart'), dateFormat),
+            after = moment(this.model.get('dateEnd'), dateFormat),
+            isValid = false;
+
+        if (!before || !after) {
```

Comment:
Looks like the validation message is behaving strangely in your screenshot, I'll revist and make sure I didn't alter something as I was cleaning up my commits.